src/contrib/cluster/Akka.Cluster.Tools/Client/ClusterClient.cs

```diff
@@ -277,33 +326,35 @@ public static Props Props(ClusterClientSettings settings)
         return Actor.Props.Create(() => new ClusterClient(settings)).WithDeploy(Deploy.Local);
     }
 
-    private ILoggingAdapter _log = Context.GetLogger();
+    private readonly ILoggingAdapter _log = Context.GetLogger();
+    private readonly EventStream _eventStream = Context.System.EventStream;
     private readonly ClusterClientSettings _settings;
     private readonly DeadlineFailureDetector _failureDetector;
-    private ImmutableHashSet<ActorPath> _contactPaths;
-    private readonly ActorSelection[] _initialContactsSelections;
-    private ActorSelection[] _contacts;
-    private ImmutableHashSet<ActorPath> _contactPathsPublished;
```

Comment:
Combined `ActorPath` and `ActorSelection` tracking into the `Contact` class so we can track which contacts to remove from the contact list when a contact is down